testsuite/MDAnalysisTests/parallelism/test_pickle_transformation.py

```diff
@@ -151,3 +151,17 @@ def test_add_unwrap_pickle(unwrap_transformation, u):
     u.trajectory[0]
     for u_ts, u_p_ts in zip(u.trajectory[:5], u_p.trajectory[:5]):
         assert_almost_equal(u_ts.positions, u_p_ts.positions)
+
+
+def test_atomgroup_pickle(u):
+    atm = pickle.loads(pickle.dumps(u.atoms))
+    assert_almost_equal(u.atoms.positions, atm.positions)
+
+
+def test_residuegroup_pickle(u):
+    res = pickle.loads(pickle.dumps(u.residues[10:20]))
+    assert_almost_equal(u.residues[10:20].universe.atoms.positions, res.universe.atoms.positions)
+
+def test_segmentgroup_pickle(u):
+    seg = pickle.loads(pickle.dumps(u.segments[0:]))
+    assert_almost_equal(u.segments[0:].universe.atoms.positions, seg.universe.atoms.positions)
```

Comment:
I'm not sure this is the right file, if there's not an existing suitable file you could always make `test_core_pickling.py`

Reply:
This should go into core/test_segmentgroup.py

Reply:
Fixed on #3953.